docs/docs/create-source-plugin.md

```diff
@@ -100,3 +100,54 @@ package provides a set of helper functions to generate Node objects with the
 required fields. This includes automatically generating fields like node IDs
 and the `contentDigest` MD5 hash, keeping your code focused on data gathering,
 not boilerplate.
+
+## Advanced
+
+### Adding relationships between nodes
+
+Gatsby source plugins not only create nodes, they also create relationships between nodes that are exposed to GraphQL queries.
+
+There are two ways of adding node relationships in Gatsby: (1) hierarchical (parent-child) or (2) foreign-key based. 
+
+#### Hierarchical relationships
+
+An example of a hierarchical relationship is `the gatsby-transformer-remark` plugin, which transforming a parent `fileNode`'s markdown string into a `MarkdownRemark` node. The transformer plugin adds its newly created child node as a child to the parent node using the action `createParentChildLink`. The markdown node is completely derived from the parent File node so wouldn't ever exist if the parent `fileNode` hadn't been created.
```

Comment:
`the gatsby-transformer-remark` - should "the" be in backticks?

Reply:
fixed thx